libmount/src/utils.c

```diff
@@ -1342,6 +1345,210 @@ int mnt_tmptgt_cleanup(int old_ns_fd)
 #endif
 }
 
+#ifdef UL_HAVE_MOUNT_API
+static int write_id_mapping(idmap_type_t idtype, pid_t pid, const char *buf,
+			    size_t buf_size)
+{
+	int fd = -1, ret = -1, setgroups_fd = -1;
+	char path[PATH_MAX];
+
+	if (geteuid() != 0 && idtype == ID_TYPE_GID) {
+		snprintf(path, sizeof(path), "/proc/%d/setgroups", pid);
+
+		setgroups_fd = open(path, O_WRONLY);
```

Comment:
in the systemd codebase at east we have the strict rule that all open() calls have to take place with O_CLOEXEC. 

And this is library code, so you never know if maybe it's called from a threaded environment, where not specifying O_CLOEXEC would hurt?

Reply:
Ah yeah, oversight on my part, thanks!